internal/eval/store_eval.go

```diff
@@ -600,3 +600,258 @@ func evalPFMERGE(args []string, store *dstore.Store) *EvalResponse {
+		Result: clientio.OK,
+		Error:  nil,
+	}
+}
+
+// evalHMGET returns an array of values associated with the given fields,
+// in the same order as they are requested.
+// If a field does not exist, returns a corresponding nil value in the array.
+// If the key does not exist, returns an array of nil values.
+func evalHMGET(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 2 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("HMGET"),
+		}
+	}
+	key := args[0]
+
+	// Fetch the object from the store using the key
+	obj := store.Get(key)
+
+	// Initialize the results slice
+	results := make([]interface{}, len(args[1:]))
+
+	// If the object is nil, return empty results for all requested fields
+	if obj == nil {
+		for i := range results {
+			results[i] = nil // Return nil for non-existent fields
+		}
+		return &EvalResponse{
+			Result: results,
+			Error:  nil,
+		}
+	}
+
+	// Assert that the object is of type HashMap
+	if err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeHashMap, object.ObjEncodingHashMap); err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongTypeOperation,
+		}
+	}
+
+	hashMap := obj.Value.(HashMap)
+
+	// Loop through the requested fields
+	for i, hmKey := range args[1:] {
+		hmValue, present := hashMap.Get(hmKey)
+		if present {
+			results[i] = *hmValue // Set the value if it exists
+		} else {
+			results[i] = nil // Set to nil if field does not exist
+		}
+	}
+
+	// Return the results and no error
+	return &EvalResponse{
+		Result: results,
+		Error:  nil,
+	}
+}
+
+func evalHGET(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) != 2 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("HGET"),
+		}
+	}
+
+	key := args[0]
+	hmKey := args[1]
+
+	val, err := getValueFromHashMap(key, hmKey, store)
+	if err != nil {
+		return &EvalResponse{
+			Error: err,
+		}
+	}
+
+	if val == nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  nil,
+		}
+	}
+
+	return &EvalResponse{
+		Result: *val,
+		Error:  nil,
+	}
+}
+
+func evalHSETNX(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) != 3 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("HSETNX"),
+		}
+	}
+
+	key := args[0]
+	hmKey := args[1]
+
+	val, err := getValueFromHashMap(key, hmKey, store)
+	if err != nil {
+		return &EvalResponse{
+			Error: err,
+		}
+	}
+
+	if val != nil {
+		return &EvalResponse{
+			Result: int64(0),
+			Error:  nil,
+		}
+	}
+
+	evalHSET(args, store)
+
+	return &EvalResponse{
+		Result: int64(1),
+	}
+}
+
+func evalHDEL(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 2 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("HDEL"),
+		}
+	}
+
+	key := args[0]
+	fields := args[1:]
+
+	obj := store.Get(key)
+
+	if obj == nil {
+		return &EvalResponse{
+			Result: int64(0),
```

Comment:
Although the zero value of an error will be nil, it is still better to assign it for consistency.